platform/platform-scheduler/src/main/java/ddf/platform/scheduler/CommandJob.java

```diff
@@ -53,132 +55,132 @@ protected Subject getSystemSubject() {
+        if (session != null) {
+          LOGGER.trace("Executing command \"{}\"", commandString);
+          try {
+            session.execute(commandString);
+
+            try {
+              final String commandOutput =
+                  byteArrayOutputStream.toString(StandardCharsets.UTF_8.name());
+
+              LOGGER.trace("Execution output for command \"{}\": {}", commandString, commandOutput);
+            } catch (UnsupportedEncodingException e) {
+              LOGGER.debug("Unable to get command output.", e);
+            }
+          } catch (Exception e) {
+            printErrorMessage(commandString);
+            LOGGER.debug("Unable to execute command.", e);
+          }
+        } else {
+          printErrorMessage(commandString);
+          LOGGER.debug("Unable to create session.");
         }
+      } catch (IOException e) {
+        printErrorMessage(commandString);
+        LOGGER.debug("Unable to create session.", e);
       }
-    } catch (UnsupportedEncodingException e) {
-      LOGGER.info("Unable to produce output", e);
-    } catch (IOException e) {
-      LOGGER.warn(
-          "Error with the emptyInputStream used as a work-around for a Karaf bug. This should never happen. There is an issue with the Platform Command Scheduler implementation.",
-          e);
-    } finally {
-      if (session != null) {
-        session.close();
-      }
+    } else {
+      printErrorMessage(commandString);
+      LOGGER.debug("Unable to create session factory.");
     }
   }
 
-  private PrintStream getPrintStream(ByteArrayOutputStream byteArrayOutputStream)
+  private PrintStream createPrintStream(ByteArrayOutputStream byteArrayOutputStream)
       throws UnsupportedEncodingException {
     return new PrintStream(byteArrayOutputStream, false, StandardCharsets.UTF_8.name());
   }
 
-  private String checkInput(JobExecutionContext context) throws CommandException {
-    String command = null;
+  @Nullable
+  private static String getCommandString(JobExecutionContext context) {
+    if (context != null) {
+      final JobDataMap mergedJobDataMap = context.getMergedJobDataMap();
 
-    if (context == null) {
-      LOGGER.debug(
-          "No JobExecutionContext found. Could not fire {}", CommandJob.class.getSimpleName());
-      throw new CommandException();
-    }
-
-    JobDataMap mergedJobDataMap = context.getMergedJobDataMap();
-
-    if (mergedJobDataMap == null) {
-      LOGGER.debug("No input found. Could not fire {}", CommandJob.class.getSimpleName());
-      throw new CommandException();
-    }
-
-    if (mergedJobDataMap.getString(COMMAND_KEY) != null) {
-      command = mergedJobDataMap.getString(COMMAND_KEY);
+      if (mergedJobDataMap != null) {
+        return mergedJobDataMap.getString(COMMAND_KEY);
+      }
     }
 
-    return command;
+    return null;
   }
 
-  private static class CommandException extends Exception {}
+  private void printErrorMessage(String commandString) {
```

Comment:
❓ Should we log a `warn` instead of `info`? This may be something the admin can fix and is fairly important.
✏️  Nit: Since this is really logging a message and not printing to the console, should we name it `logInfoMessage()` or `logWarnMessage()` instead?